code/obj/machinery/singularity.dm

```diff
@@ -26,11 +26,16 @@ Contains:
 	..()
 */
 /obj/machinery/the_singularitygen/process()
-	var/checkpointC = 0
-	for (var/obj/X in orange(4,src))
-		if (istype(X, /obj/machinery/containment_field))
-			checkpointC ++
-	if (checkpointC >= 20)
+	var/goodgenerators = 0 //ensures that there are 4 generators in place with at least 2 links. note that false positives are very possible and will result in a loose singularity
+	var/smallestdimension = 0//determines the radius of the produced singularity
+	for (var/obj/X in orange(11,src))
```

Comment:
why the change

Reply:
The change was necessary to establish the size of the singularity that would spawn as that is no longer constant.

Reply:
`for (var/obj/X in orange(11,src))` 😱 

just... no. Use by_type or something instead